### ISSUE TITLE
Negative areas found

### ISSUE THREAD

**Issue by Rizhiy** [roytseng-tw/Detectron.pytorch#7]
When I try to run the training code, I get the following error:
```
RuntimeWarning: Negative areas found: 3
```

I'm running: `e2e_mask_rcnn_R-101-FPN_2x.yaml`

**Comment by roytseng-tw** [comment]
This is a warning not error. Generally, you can ignore it.
Rois predicted by RPN with negative areas will be excluded from the following process.

**Comment by MFattouh** [comment]
Ignoring the negative areas lean to NaN mask loss and the total loss becomes NaN as well. IMHO, the current batch should be discarded in this case. Any hint on how to do so?!

**Comment by MFattouh** [comment]
Answering my question, I did a simple NaN check before the weights update and discard the current batch.
My code looks like this:
`` 
loss = net_outputs['total_loss']
if torch.isnan(loss):
    logger.info('NaN loss found! Skipping the current step ...')
    break
loss.backward()
``

**Comment by PkuRainBow** [comment]
@MFattouh Hi, I also meet the case that loss become NAN after caught this warning.

So I am wondering how you fix this problem?

**Comment by PkuRainBow** [comment]
@roytseng-tw @Rizhiy Could you share me how to solve this problem?

**Comment by Rizhiy** [comment]
I was only using the detection part of the network, so NaN loss was not a problem for me.
